app/models/miq_server.rb

```diff
@@ -33,6 +33,8 @@ class MiqServer < ApplicationRecord
 
   scope :with_zone_id, ->(zone_id) { where(:zone_id => zone_id) }
 
+  validates :hostname, :allow_nil => true, :format => {:with => MoreCoreExtensions::StringFormats::RE_HOSTNAME, :multiline => true}
```

Comment:
I think this will fail save calls for unrelated changes if we have a record in the database with an invalid hostname already.

I would rather have a migration that removes invalid hostnames from the miq_servers table.